app/helpers/application_helper.rb

```diff
@@ -84,6 +61,41 @@ def origin_display_name(practice)
     end
   end
 
+  def generate_facility_names(practice)
+    fac_type = Practice.initiating_facility_types[practice.initiating_facility_type]
+    locs = practice.practice_origin_facilities.includes(determine_associations(practice)).where(facility_type: fac_type)
+    locs.map { |loc| construct_facility_name(loc) }.join(', ')
+  end
+
+  def determine_associations(practice)
+    associations = [:va_facility]
+    if practice.practice_origin_facilities.where(va_facility: nil).exists?
+      associations << :clinical_resource_hub
+    end
+    associations
+  end
+
+  def construct_facility_name(loc)
+    has_va_facility = loc.va_facility.present?
+    official_station_name = has_va_facility ? loc.va_facility.official_station_name : loc.clinical_resource_hub.official_station_name
+    common_name = loc.va_facility.common_name if has_va_facility
+    has_va_facility ? facility_name_with_common_name(official_station_name, common_name) : official_station_name
+  end
```

Comment:
while we're here, could this be simplified to an if / else on `va.facility.present?` ?